#!/usr/bin/env python
# set up dynamically reconfigurable parameters here 
PACKAGE = "waypoint_updater"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("dyn_test_stoplight_wp",   int_t, 0, "TESTING next stopline wp",
        -1, -1, 11010)
gen.add("dyn_update_rate",         int_t, 0, "rate (Hz) at which final waypoints published",
        30, 1, 50)
gen.add("dyn_lookahead_wps",       int_t, 0, "number of final waypoints to send",
        40, 10,  100)
gen.add("dyn_default_velocity", double_t, 0, "top speed to use (mps)",
        17.0, 0.0, 100.0)
gen.add("dyn_default_accel",    double_t, 0, "accel/decel rate (m/s) to use",
        0.8, 0.5, 5.0)
gen.add("dyn_tl_buffer",        double_t, 0, "buffer distance (m) to stop at before traffic light wp",
        3.0, 0.0, 5.0)
gen.add("dyn_buffer_offset",    double_t, 0, "offset into buffer to target stopping point",
        2.5, 0.0, 5.0)
gen.add("dyn_creep_zone",       double_t, 0, "zone before stoplight where no speedup beyond creep",
        20.0, 0.0, 35.0)
# gen.add("dyn_jmt_time_factor",  double_t, 0, "factor to extend jmt accel / decel time ",
#         1.0, 0.0, 1.8)
        
exit(gen.generate(PACKAGE, "waypoint_updater", "DynReconf"))
